pkg/cli/powertools/changestateintospec/cmd.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package changestateintospec
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/cli/powertools/diffs"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/cli/powertools/kubecli"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/stateintospec"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
+)
+
+// Options configures the behaviour of the ChangeStateIntoSpec operation.
+type Options struct {
+	kubecli.ClusterOptions
+	kubecli.ObjectOptions
+
+	// NewStateIntoSpecAnnotation is the new value for the state-into-spec annotation
+	NewStateIntoSpecAnnotation string
+
+	// FieldOwner is the field-manager owner value to use when making changes
+	FieldOwner string
+
+	// DryRun is true if we should not actually make changes, just print the changes we would make
+	DryRun bool
+
+	// PreserveFields is a list of additional spec fields we should preserve
+	PreserveFields []string
+}
+
+func (o *Options) PopulateDefaults() {
+	o.ClusterOptions.PopulateDefaults()
+	o.ObjectOptions.PopulateDefaults()
+
+	o.FieldOwner = "change-state-into-spec"
+	o.NewStateIntoSpecAnnotation = "absent"
+	o.DryRun = false
```

Comment:
We decided no, because long-term we want the default to be dry-run=false, and if we started with default is dry-run=true it would be hard to go to dry-run=false.